src/control/Control.Layers.js

```diff
@@ -373,7 +373,7 @@ export var Layers = Control.extend({
 			layer = this._getLayer(input.layerId).layer;
 			hasLayer = this._map.hasLayer(layer);
 
-			if (input.checked && !hasLayer) {
+			if (input.checked) {
 				addedLayers.push(layer);
 
 			} else if (!input.checked && hasLayer) {

```

Comment:
Even if not strictly necessary, could you move this `hasLayer` check down to the layer removal, so that it works the same way as the loop for adding layers? Easier to understand the code when it is symmetric.